spec/services/reporter_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Reporter do
+  let!(:test_start_time) { DateTime.now } # useful for both output cleanup and CSV filename testing
+
+  let!(:msr_a) { create(:moab_storage_root) }
+  let!(:complete_moab_1) { create(:complete_moab, moab_storage_root: msr_a) }
+  let!(:complete_moab_2) { create(:complete_moab, moab_storage_root: msr_a) }
+  let!(:complete_moab_3) { create(:complete_moab, moab_storage_root: msr_a) }
+
+  let!(:msr_b) { create(:moab_storage_root) }
+  let!(:complete_moab_4) { create(:complete_moab, moab_storage_root: msr_b) }
+  let!(:complete_moab_5) { create(:complete_moab, moab_storage_root: msr_b) }
+  let!(:complete_moab_6) { create(:complete_moab, moab_storage_root: msr_b) }
+
+  after do
+    next unless FileTest.exist?(described_class.default_filepath)
+    Dir.each_child(described_class.default_filepath) do |filename|
+      fullpath_filename = File.join(described_class.default_filepath, filename)
+      File.unlink(fullpath_filename) if File.stat(fullpath_filename).mtime > test_start_time
+    end
+  end
+
+  describe '.moab_storage_root_druid_list_to_csv' do
+    it 'creates a file containing a list of druids from the given storage root' do
+      csv_filename = described_class.moab_storage_root_druid_list_to_csv(storage_root_name: msr_b.name)
+      expect(CSV.read(csv_filename)).to eq(
+        [complete_moab_4, complete_moab_5, complete_moab_6].map { |cm| [cm.preserved_object.druid] }
+      )
+    end
+
+    it 'names the file with the default prefix and a timestamp and writes it to the default location' do
+      # file timestamp is to the second, but test_start_time is more precise.  this sleep allows us to compare
+      # timestamp to test start time without having to round fractions of a second.
+      sleep(1.second)
```

Comment:
i can see how this test would be prone to flapping, though it hadn't happened to me before.  but it consistently happened once i made the change @jcoyne suggested in [his review](https://github.com/sul-dlss/preservation_catalog/pull/1347#discussion_r373027708).  i assume it's something like the `let!` getting instantiated later (closer to when the file is written) compared to the constant it replaced.

anyway, hopefully this approach to mitigating the problem described in this comment is acceptable -- the alternatives were to either not have the expectation about the time stamp being greater than the start time of the tests, or to round fractions of a second off the start time, and those both struck me as less appealing options than the (very short) `sleep` here.  though i do understand that this is a pattern we want to use as sparingly as possible.